### PR TITLE
YJ_003: Create Project Endpoint

### PR DESCRIPTION
**Pull request by dylanBkainos**
👍

### CODE REVIEW

File: src/main/java/org/example/daos/ProjectDao.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                        projectIdResultSet.getBoolean("completed"));
+            }
+        }
+        return null;
+    }
+
+    public void updateProject(final int projectId,
+                              final ProjectRequest projectRequest)
+            throws SQLException {
+        Connection connection = DatabaseConnector.getConnection();
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/ProjectDao.java]: try around this connection pls

File: src/main/java/org/example/daos/ProjectDao.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                connection.prepareStatement(updateStatement);
+        statement.setBoolean(1, projectRequest.getCompleted());
+        statement.setInt(2, projectId);
+
+        statement.executeUpdate();
+    }
+
+    public int createProject(final ProjectRequest projectRequest)
+            throws SQLException {
+        Connection c = DatabaseConnector.getConnection();
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/ProjectDao.java]: surround with a try

File: src/main/java/org/example/daos/ProjectDao.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+                        + "techLeadId,"
+                        + "salesEmployeeId) VALUES (?, ?, ?, ?, ?)";
+        PreparedStatement st = c.prepareStatement(selectStatement,
+                Statement.RETURN_GENERATED_KEYS);
+
+        final int firstValue = 1;
+        final int secondValue = 2;
+        final int thirdValue = 3;
+        final int fourthValue = 4;
+        final int fifthValue = 5;
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/ProjectDao.java]: please dont just say values, instead do what they are

File: src/main/java/org/example/daos/ProjectDao.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                }
+
+                Project project = new Project(
+                        projectId,
+                        resultSet.getString("name"),
+                        resultSet.getDouble("projectValue"),
+                        resultSet.getInt("clientId"),
+                        deliveryEmployees,
+                        resultSet.getInt("techLeadId"),
+                        resultSet.getBoolean("completed"));
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/ProjectDao.java]: this code is duplicated

File: src/main/java/org/example/exceptions/DoesNotExistException.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package org.example.exceptions;
+
+public class DoesNotExistException extends Throwable {
```
Comment (lukemKainos) [inline src/main/java/org/example/exceptions/DoesNotExistException.java]: Extend exception

File: src/main/java/org/example/mappers/ProjectMapper.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        return projects.stream()
+                .map(project -> new ProjectResponse(
+                        project.getProjectId(),
+                        project.getName(),
+                        project.getProjectValue(),
+                        project.getClientId(),
+                        project.getDeliveryEmployeeId(),
+                        project.getTechLeadId(),
+                        project.getCompleted()))
+                .collect(Collectors.toList());
```
Comment (lukemKainos) [inline src/main/java/org/example/mappers/ProjectMapper.java]: return projects.stream()
                .map(ProjectMapper::mapToProjectResponse)
                .collect(Collectors.toList());

File: src/main/java/org/example/models/Project.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+import java.util.List;
+
+public class Project {
+    private int projectId;
+    private String name;
+    private double projectValue;
+    private int clientId;
+    private List<Integer> deliveryEmployeeId;
+    private int techLeadId;
+    private boolean completed;
```
Comment (lukemKainos) [inline src/main/java/org/example/models/Project.java]: should be isCompleted

**Review by lukemKainos — CHANGES_REQUESTED** [review]

**Review by Aaron-Millen — APPROVED** [review]

**Review by Aaron-Millen — APPROVED** [review]

**Review by lukemKainos — APPROVED** [review]

**Review by lukemKainos — APPROVED** [review]

**Pull request merged**